src/iptvsimple/utilities/StreamUtils.cpp

```diff
@@ -63,7 +63,37 @@ void StreamUtils::SetAllStreamProperties(PVR_NAMED_VALUE* properties, unsigned i
     }
     else // inputstream.adpative
     {
-      StreamUtils::SetStreamProperty(properties, propertiesCount, propertiesMax, PVR_STREAM_PROPERTY_STREAMURL, streamURL);
+      // If no media headers are explicitly set for inputstream.adaptive,
+      // strip the headers from streamURL and put it to media headers property
+
+      bool haveStreamHeaders = !channel.GetProperty("inputstream.adaptive.stream_headers").empty();
+      if (!haveStreamHeaders)
+      {
+        if (StreamUtils::HasProtocolOptions(streamURL))
```

Comment:
My bad. Providing this function doesn't add any efficiency. It just means you have to do the find twice. Sorry. So you should go back to the `size_t found =` above the if statement and found in the if.

Sorry again!